rotkehlchen/chain/ethereum/manager.py

```diff
@@ -619,53 +589,24 @@ def _get_code(self, web3: Optional[Web3], account: ChecksumEthAddress) -> str:
 
     def ens_reverse_lookup(self, reversed_addresses: List[ChecksumEthAddress]) -> Dict[ChecksumEthAddress, Optional[str]]:  # noqa: E501
         """Performs a reverse ENS lookup on a list of addresses
-
-        Because a multicall is used, no exceptions are raised.
-        If any exceptions occur, they are logged and None is returned for that
-        """
-        human_names: Dict[ChecksumEthAddress, Optional[str]] = {}
-        # Querying resolvers' addresses
-        resolver_params = [
-            EnsContractParams(address=addr, abi=ENS_ABI, method_name='resolver', arguments=_prepare_ens_call_arguments(addr))  # noqa: E501
-            for addr in reversed_addresses
-        ]
-        resolvers_output = multicall(
-            ethereum=self,
-            calls=[(ENS_MAINNET_ADDR, _encode_ens_contract(params=params)) for params in resolver_params],  # noqa: E501
-        )
-        resolvers = []
-        # We need a new list for reversed_addresses because not all addresses have resolver
-        filtered_reversed_addresses = []
-        # Processing resolvers query output
-        for reversed_addr, params, resolver_output in zip(reversed_addresses, resolver_params, resolvers_output):  # noqa: E501
-            decoded_resolver = _decode_ens_contract(params=params, result_encoded=resolver_output)
-            if decoded_resolver is None:
-                human_names[reversed_addr] = None
-                continue
-            try:
-                deserialized_resolver = deserialize_ethereum_address(decoded_resolver)
-            except DeserializationError:
-                log.error(
-                    f'Error deserializing address {decoded_resolver} while doing reverse ens lookup',  # noqa: E501
-                )
-                human_names[reversed_addr] = None
-                continue
-            resolvers.append(deserialized_resolver)
-            filtered_reversed_addresses.append(reversed_addr)
-
-        # Querying human names
-        human_names_params = [
-            EnsContractParams(address=resolver, abi=ENS_RESOLVER_ABI, method_name='name', arguments=_prepare_ens_call_arguments(addr))  # noqa: E501
-            for addr, resolver in zip(filtered_reversed_addresses, resolvers)]
-        human_names_output = multicall(
+        May raise:
+        - RemoteError if etherscan is used and there is a problem with
+        reaching it or with the returned result
+        - BlockchainQueryError if web3 is used and there is a VM execution error"""
+        for addr in reversed_addresses:
+            if is_checksum_address(addr) is False:
+                raise InputError(f'One of the addresses {addr} is not a valid checksum address')
+        result = ENS_REVERSE_RECORDS.call(
             ethereum=self,
-            calls=[(params.address, _encode_ens_contract(params=params)) for params in human_names_params],  # noqa: E501
+            method_name='getNames',
+            arguments=[reversed_addresses],
```

Comment:
Also can you change the name of this argument? It's confusing. The addresses are not reversed. It's the addresses whose reverse ens names you want to find. So perhaps call them `queried_addresses` or even simply `addresses`?

Reply:
Yeah, sounds reasonable, changed to `addresses`